### PR TITLE
MONGOCRYPT-545 Implement FLE2FindRangePayloadV2 transform

### PR DESCRIPTION
**Pull request by sgolemon**
As with MONGOCRYPT-544, this is nice and straightforward.

### CODE REVIEW

File: src/mongocrypt-marking.c [hunk 1]
```diff
@@ -1528,9 +1829,113 @@ _mongocrypt_fle2_placeholder_to_find_ciphertextForRange (
...
+   // Serialize.
+   {
+      bson_t out = BSON_INITIALIZER;
+      mc_FLE2FindRangePayloadV2_serialize (&payload, &out);
+      _mongocrypt_buffer_steal_from_bson (&ciphertext->data, &out);
+   }
+   _mongocrypt_buffer_steal (&ciphertext->key_id, &placeholder->index_key_id);
+
+   // Do not set ciphertext->original_bson_type and ciphertext->key_id. They are
+   // not used for FLE2FindRangePayload.
```
Comment (kevinAlbs) [inline src/mongocrypt-marking.c:1929]:
```suggestion
   // not used for FLE2FindRangePayloadV2.
```

**Review by kevinAlbs — APPROVED** [review]
LGTM. The separate smaller PRs are appreciated for reviewing.

**Review by markbenvenuto — APPROVED** [review]

File: src/mongocrypt-marking.c (hunk 1 above)
Comment (sgolemon) [reply src/mongocrypt-marking.c]: 👍

**Review by erwee — APPROVED** [review]

**Comment by sgolemon** [comment]
Merged as 2ec9c3c4ff94f76bb51a5b4cc2fe9190457d6541

**Pull request closed without merging**